modules/atmosphere/rendering/atmospheredeferredcaster.cpp

```diff
@@ -1190,369 +1000,565 @@ void AtmosphereDeferredcaster::executeCalculations(GLuint quadCalcVAO,
                         << static_cast<unsigned int>(pixels[k + 2]) << " ";
-                k += 3;
+                    k += 3;
+                }
+                ppmFile << std::endl;
             }
-            ppmFile << std::endl;
-        }
-        delete[] pixels;
+            delete[] pixels;
 
-        ppmFile.close();
-    }
-}
-
-bool AtmosphereDeferredcaster::isAtmosphereInFrustum(const double* MVMatrix,
-                                                     const glm::dvec3& position,
-                                                     double radius) const
-{
-
-    // Frustum Planes
-    glm::dvec3 col1(MVMatrix[0], MVMatrix[4], MVMatrix[8]);
-    glm::dvec3 col2(MVMatrix[1], MVMatrix[5], MVMatrix[9]);
-    glm::dvec3 col3(MVMatrix[2], MVMatrix[6], MVMatrix[10]);
-    glm::dvec3 col4(MVMatrix[3], MVMatrix[7], MVMatrix[11]);
-
-    glm::dvec3 leftNormal = col4 + col1;
-    glm::dvec3 rightNormal = col4 - col1;
-    glm::dvec3 bottomNormal = col4 + col2;
-    glm::dvec3 topNormal = col4 - col2;
-    glm::dvec3 nearNormal = col3 + col4;
-    glm::dvec3 farNormal = col4 - col3;
-
-    // Plane Distances
-    double leftDistance   = MVMatrix[15] + MVMatrix[12];
-    double rightDistance  = MVMatrix[15] - MVMatrix[12];
-    double bottomDistance = MVMatrix[15] + MVMatrix[13];
-    double topDistance    = MVMatrix[15] - MVMatrix[13];
-    double nearDistance   = MVMatrix[15] + MVMatrix[14];
-    double farDistance    = MVMatrix[15] - MVMatrix[14];
-
-    // Normalize Planes
-    double invMag = 1.0 / glm::length(leftNormal);
-    leftNormal   *= invMag;
-    leftDistance *= invMag;
-
-    invMag = 1.0 / glm::length(rightNormal);
-    rightNormal   *= invMag;
-    rightDistance *= invMag;
-
-    invMag = 1.0 / glm::length(bottomNormal);
-    bottomNormal   *= invMag;
-    bottomDistance *= invMag;
-
-    invMag = 1.0 / glm::length(topNormal);
-    topNormal   *= invMag;
-    topDistance *= invMag;
-
-    invMag = 1.0 / glm::length(nearNormal);
-    nearNormal   *= invMag;
-    nearDistance *= invMag;
-
-    invMag = 1.0 / glm::length(farNormal);
-    farNormal   *= invMag;
-    farDistance *= invMag;
-
-    if ((glm::dot(leftNormal, position) + leftDistance) < -radius) {
-        return false;
-    }
-    else if ((glm::dot(rightNormal, position) + rightDistance) < -radius) {
-        return false;
-    }
-    else if ((glm::dot(bottomNormal, position) + bottomDistance) < -radius) {
-        return false;
-    }
-    else if ((glm::dot(topNormal, position) + topDistance) < -radius) {
-        return false;
-    }
-    else if ((glm::dot(nearNormal, position) + nearDistance) < -radius) {
-        return false;
+            ppmFile.close();
+        }
     }
-    // The far plane testing is disabled because the atm has no depth.
-    /*else if ((glm::dot(farNormal, position) + farDistance) < -radius) {
+
+    bool AtmosphereDeferredcaster::isAtmosphereInFrustum(const double* MVMatrix,
```

Comment:
Alternative to raw pointer;   better use a `glm::dmat4x4`?

Reply:
Done.